File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/model/sharedpref/TokenSpModel.java
package uk.co.victoriajanedavis.chatapp.data.model.sharedpref;

import java.util.UUID;


public class TokenSpModel {

    public static final String PREF_TOKEN_KEY = "tokenKey";
    public static final String PREF_USER_UUID_KEY = "userUuidKey";
    public static final String PREF_USER_USERNAME_KEY = "userUsernameKey";
    public static final String PREF_USER_EMAIL_KEY = "userEmailKey";

    private String token;
    private UUID userUuid;
    private String userUsername;
    private String userEmail;


    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public UUID getUserUuid() {
        return userUuid;
    }

    public void setUserUuid(UUID userUuid) {
        this.userUuid = userUuid;
    }

    public String getUserUsername() {
        return userUsername;
    }

    public void setUserUsername(String userUsername) {
        this.userUsername = userUsername;
    }

    public String getUserEmail() {
        return userEmail;
    }

    public void setUserEmail(String userEmail) {
        this.userEmail = userEmail;
    }
}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/model/sharedpref/TokenSpModel.kt
package uk.co.victoriajanedavis.chatapp.data.model.sharedpref

import java.util.UUID


data class TokenSpModel(
    val token: String,
    val userUuid: UUID? = null,
    val userUsername: String,
    val userEmail: String
) {

    companion object {
        const val PREF_TOKEN_KEY = "tokenKey"
        const val PREF_USER_UUID_KEY = "userUuidKey"
        const val PREF_USER_USERNAME_KEY = "userUsernameKey"
        const val PREF_USER_EMAIL_KEY = "userEmailKey"
    }
}
